be/src/pipeline/exec/exchange_sink_buffer.cpp

```diff
@@ -45,16 +45,40 @@
 #include "vec/sink/vdata_stream_sender.h"
 
 namespace doris {
+
 namespace vectorized {
+BroadcastPBlockHolder::~BroadcastPBlockHolder() {
+    // lock the parent queue, if the queue could lock success, then return the block
+    // to the queue, to reuse the block
+    std::shared_ptr<BroadcastPBlockHolderQueue> tmp_queue = _parent_creator.lock();
+    if (tmp_queue != nullptr) {
+        tmp_queue->push(BroadcastPBlockHolder::create_shared(std::move(_pblock)));
+    }
+    // If the queue already deconstruted, then release pblock automatically since it
+    // is a unique ptr.
+}
 
-void BroadcastPBlockHolder::unref() noexcept {
-    DCHECK_GT(_ref_count._value, 0);
-    auto old_value = _ref_count._value.fetch_sub(1);
-    if (_dep && old_value == 1) {
-        _dep->return_available_block();
+void BroadcastPBlockHolderQueue::push(std::shared_ptr<BroadcastPBlockHolder> holder) {
+    std::unique_lock l(_holders_lock);
+    holder->set_parent_creator(shared_from_this());
+    _holders.push(holder);
+    if (_broadcast_dependency) {
+        _broadcast_dependency->set_ready();
     }
 }
 
+std::shared_ptr<BroadcastPBlockHolder> BroadcastPBlockHolderQueue::pop() {
+    std::unique_lock l(_holders_lock);
+    if (_holders.empty()) {
+        return {};
+    }
+    std::shared_ptr<BroadcastPBlockHolder> res = _holders.top();
+    _holders.pop();
+    if (_holders.empty() && _broadcast_dependency != nullptr) {
```

Comment:
DCHECK(_broadcast_dependency != nullptr);

Reply:
C ould not add this decheck ,because pipeline also use this code.